girder/api/sftp.py

```diff
@@ -55,10 +55,7 @@ def wrapped(*args, **kwargs):
 
 
 def _getFileSize(file):
-    if file.get('assetstoreId'):
-        return file['size']
-    elif file.get('linkUrl'):
-        return len(file['linkUrl'])
+    return file.get('size', len(file.get('linkUrl', '')))
```

Comment:
It looks like with the new model API, link files will *always* have a ``size`` field that defaults to 0, so this will always return 0 if none is passed. Is that the desired behavior? It seems like it might be better to only put a ``size`` field in the document if it was explicitly passed, and change the default value in the signature to ``None``.